src/api/process.c

```diff
@@ -263,17 +263,21 @@ static int process_start(lua_State* L) {
           dup2(self->child_pipes[new_fds[stream]][new_fds[stream] == STDIN_FD ? 0 : 1], stream);
         close(self->child_pipes[stream][stream == STDIN_FD ? 1 : 0]);
       }
-      if ((!detach || setsid() != -1) && (!cwd || chdir(cwd) != -1))
-        execvp((const char*)cmd[0], (char* const*)cmd);  
+      int set;
+      for (set = 0; set < env_len && setenv(env_names[set], env_values[set], 1) == 0; ++set);
```

Comment:
`setenv` can fail, we should probably error out in that case.

Reply:
We do.

Reply:
Ah, yes, you're right.